x-pack/plugins/security/server/saved_objects/saved_objects_security_extension.ts

```diff
@@ -158,6 +159,30 @@ export class SavedObjectsSecurityExtension implements ISavedObjectsSecurityExten
     auditCallback?.();
   }
 
+  async performAuthorization<A extends string>(
+    params: PerformAuthorizationParams<A>
+  ): Promise<CheckAuthorizationResult<A>> {
+    const checkResult: CheckAuthorizationResult<A> = await this.checkAuthorization({
+      types: params.types,
+      spaces: params.spaces,
+      actions: params.actions,
+      options: { allowGlobalResource: params.options?.allowGlobalResource === true },
+    });
+
+    if (params.enforceMap !== undefined && checkResult) {
+      params.actions.forEach((action) => {
+        this.enforceAuthorization({
+          typesAndSpaces: params.enforceMap!,
```

Comment:
__nit:__ to avoid `!` (or just destruct `params` entirely at line 163)
```suggestion
    const typesAndSpaces = params.enforceMap;
    if (typesAndSpaces !== undefined && checkResult) {
      params.actions.forEach((action) => {
        this.enforceAuthorization({
          typesAndSpaces,
```